reactkarma/reactkarma.py

```diff
@@ -191,6 +191,17 @@ async def _check_reaction(
         emoji = reaction.emoji
         upvote = await self._is_upvote(guild, emoji)
         if upvote is not None:
+            message = "{0}({1})".format(user.display_name, user.name)
+
+            if upvote == added:
+                message += " gave a karma to "
+            else:
+                message += " took a karma from "
+            
+            message += "{0}({1})".format(author.display_name, author.name)
+            message += "."
+            
+            await reaction.message.channel.send(message)
```

Comment:
I'm thinking, it might be better to send the message in `_add_karma()` after the karma amount has been set. This way, if the karma update fails, there's no message that is sent out.

Basically, you'd pass everything you need to `_add_karma` as additional parameters and execute the send message there

Reply:
I have hopefully learnt how async/await works.

I wanted to ask why not just place message logic after `_add_karma` call instead of placing it inside the `_add_karma `definition?

the `_add_karma` would execute first and if error happened the whole thing would close and the message would not be printed

I have checked that by adding a `raise` error statement in `_add_karma` definition and placing message logic as mentioned before and when running code, the error occurred and the message didn't get printed. 

Is the reason to reduce complexity of the `check_reaction` function as it relatively has more lines of code and logic than `_add_karma`?
Is it because it is more meaningful to add printing logic in `_add_karma` function?
Or is there any other reason for that?

Reply:
@ahmedsiam0 it's definitely better for `_add_karma()` to only handle incrementing or decrementing the count, 100% as long as it returns an error if it fails! Separation of concerns.

For now, it's fine, let's just ship this, we can improve it later.